### DISCUSSION TITLE
osgearth with Qt5 and osgqt

### DISCUSSION THREAD

**Discussion by lavi-k** [pelicanmapping/osgearth#2029] [Support Q&A]
Hello there,
I have tried to build the latest osgearth code (tried linking it against osgqt) but was not able build it.
As newer versions of osgEarth no longer have any Qt. So, I am looking for a workaround through which I can use Qt5(QOpenGLWidget) and osgqt with osgearth.
Even with an older version of osgearth will work for me.

**Comment by osgwraiESC** [comment]
on linux i'm not having any problem linking in libosgQt5.so with V3.2, my application main is based on example application:

./applications/osgearth_qt_simple/osgearth_qt_simple.cpp

which needs ViewerWidget and ViewerWidget.cpp that i think got removed at some point, but maybe you can find in the legacy versions somewhere

**Reply by lavi-k** [reply]
But ViewerWidget and ViewerWidget.cpp are missing in 3.2 branch. did you copy it from the earlier release?

**Reply by osgwraiESC** [reply]
yes i saved them from previous version, you can find them here, in the osgEarthQt folder 

https://github.com/gwaldron/osgearth/releases/tag/osgearth-2.10.2
